### ISSUE TITLE
Tour of F#: in "Pattern Matching" section, the second example code fails compilation

### ISSUE THREAD

**Issue by 2jacobtan** [dotnet/docs#18609]
![image](https://user-images.githubusercontent.com/12626411/82749627-01b20a80-9ddd-11ea-8452-bed2492eb6aa.png)

I added a type annotation to fix it.

![image](https://user-images.githubusercontent.com/12626411/82749922-2ad39a80-9ddf-11ea-90c4-b6f39c352c42.png)

Hope this helps.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 89bef555-3de1-5d81-c02b-2ac8c049423b
* Version Independent ID: eb13ad9e-2c95-979b-6d6e-13d29653c6ba
* Content: [Tour of F#](https://docs.microsoft.com/en-us/dotnet/fsharp/tour?github-auth-complete=true#feedback)
* Content Source: [docs/fsharp/tour.md](https://github.com/dotnet/docs/blob/master/docs/fsharp/tour.md)
* Product: **dotnet-fsharp**
* GitHub Login: @cartermp
* Microsoft Alias: **phcart**

**Comment by cartermp** [comment]
yoikes, thanks - this does need an update to considering .NET Core and the span overloads

**Comment by cartermp** [comment]
@gewarren @BillWagner how do I access the snippets referenced here? They're not in the dotnet/samples repo

**Comment by IEvangelist** [comment]
Hi @cartermp - this snippets are relative to the doc set. They live in the same repo as the content itself.  It is referencing the *tour.fs* file, [here](https://github.com/dotnet/docs/blob/master/samples/snippets/fsharp/tour.fs).

**Comment by BillWagner** [comment]
> how do I access the snippets referenced here? They're not in the dotnet/samples repo

@cartermp Look in the /samples folder in this repo. 

TL;DR; version: We moved the snippets into this repo to enable better workflow for build validation.  

For the long version see #17297
